### ISSUE TITLE
Accessibility quick scan 2018.rotterdam.wordcamp.org

### ISSUE THREAD

**Issue by rianrietveld** [jmslbam/wordcamp-rotterdam-2018-css#15]
**Site**: 2018.rotterdam.wordcamp.org
**Date**: February 20, 2018
**Standard**: WCAG 2 AA
**Scope**: Colours, headings, link quality and content.
**Out of scope**: 3rd party plugins and theme specific issues

**Note**: This is a quick scan, aiming to target the major issues

**TL;DR**
Major issues:
- Missing focus style in the CSS
- Some links are empty or meaningless
- Colour contrast must be improved for the green, yellow and orange

**Keyboard accessibility**
Add with `:hover` also a `:focus` style everywhere in the CSS.
That way keyboard users know where they are, while tabbing from link to link.
Tab for example trough the menu, can you see where you are?
Make the change not by a subtle change of colour, but add another visual change, like for example underlining on focus.

Link visibility
Links in the content must be visible not by colour alone.
People who are colour blind may not be able to see the links.
Quick robust fix: underline links in the content.
For example

```
.entry-content a {
        text-decoration: underline;
}
.entry-content a:hover,
.entry-content a:focus {
        text-decoration: none;
}
```

Don't set `outline: 0;`
An outline is super helpful for keyboard users with their navigation.
You can style it if you want, just don't remove it.

**Heading structure**
Heading structure is important:
One H1 per page, describing what the page is about.
The rest meaningful, sequential, don't skip a heading level
For an explanation see: [Headings in the content ](https://make.wordpress.org/accessibility/handbook/best-practices/content/using-headings-in-content/)

Heading issues:
- Home: double H1 . "Onze sponsoren" should be an H2, Gold, Silver and Bronze should be H3s.
- [Gedragscode](https://2018.rotterdam.wordcamp.org/informatie/gedragscode/): H2 is missing
- [Sponsors](https://2018.rotterdam.wordcamp.org/sponsors/) H2 is missing. Also: Add a heading H2 sponsors and give gold, silver, bronze a heading H3
- [Locatie](https://2018.rotterdam.wordcamp.org/locatie/) Use H3 sub headings for Openbaar vervoer / Watertaxi etc. Use real list items (it's now a bold dot)

**Link text**
Make link text meaningful, standout on their own and never empty. 
If you use an image as link, use the alt text as link text.
For an explanation see: [Good link texts](https://make.wordpress.org/accessibility/handbook/best-practices/content/good-link-texts/)

Link text issues
- Home: The link texts "here" or "hier" are meaningless. Suggestion: make the whole text, like "Haal hier je ticket" the link text.
- Home: The image with the tickets has no alt text, so the link is empty. Give it, for example the alt text "Haal hier je ticket".
- [Nieuws](https://2018.rotterdam.wordcamp.org/nieuws/). The thumbnail has no alt text but is a link. This results in an empty link text. Give the image the title of the post it's linking to as link text. **Note**: this maybe a theme issue
- Never use a full url as link text (screen reader users will thank you) So instead of: https://www.rotterdam.nl/wonen-leven/overzicht-p-r-terreinen/ use `overzicht P+R terreinen` as link text.

**Colour contrast**
The colour contrast should be good, so more people can read the text.
See [Use of colour](https://make.wordpress.org/accessibility/handbook/best-practices/design/use-of-color/#about-color-contrast).
There are many colour contast checkers. An easy one is [Colour Contrast Check by Snook](https://snook.ca/technical/colour_contrast/colour.html).

Colour contrast issues
The ratio must be 4.5 or higher, so Insufficient are:
- @green: #009D4e; against white : ratio = 3.539
- @green: #009D4e; against @yellow : ratio = 2.48
- @orange: #f48109; against white : ratio = 2.628

**Other issues**
An Iframe must have a title attribute, like on the contact page
`<iframe title="Blue City on Google Maps" width="100%" height="450" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" src="[...]"></iframe>`

**Some advice** (not requirements, but good practice)
- Avoid the use of capitalized sentences or short words in buttons. They are harder to read. 
- Avoid the use of centered text, also harder to read.

**Theme issues** I will not address here:
- empty links on thumbnails in the archive pages.
- order form

**Third part plugins** with issues I will not address here:
- Related posts
- JetPack
- the contact forms

@jmslbam, If you have any questions, please let me know.

**Comment by jmslbam** [comment]
Hola @rianrietveld,

It's been a while but I took the time to fix your feedback. Thank you for it! 

Here's a summary for the team.

# TLDR
Fixed everything what I could fix and what wasn't in the Theme / Plugins, except the centering of the Page title. Couldn't fined a nice visual aesthetic alternative on tablet / desktop. Did change the page titles on mobile.

-------

Site: 2019.rotterdam.wordcamp.org 
Date: Januari 21, 2019 
Standard: WCAG 2 AA 
Scope: Colours, headings, link quality and content. 



#Keyboard accessibility 

## Add with :hover also a :focus style everywhere in the CSS.
Done

 ## Tab for example trough the menu, can you see where you are? Make the change not by a subtle change of colour, but add another visual change, like for example underlining on focus.

Done: Underline + Focus + Different color.

## Link visibility Links in the content must be visible not by colour alone.

**People who are colour blind may not be able to see the links. Quick robust fix: underline links in the content.**

Done

## Don't set outline: 0; An outline is super helpful for keyboard users with their navigation. You can style it if you want, just don't remove it.

Done, enabled it back on to default behaviour.

## Heading structure 

Heading structure is important:  One H1 per page, describing what the page is about. The rest meaningful, sequential, don't skip a heading level . For an explanation see: Headings in the content 

Heading issues:

_* Home: double H1 . "Onze sponsoren" should be an H2, Gold, Silver and Bronze should be H3s._
Is wat gepuzzel ivm de werking van het Thema Twenty-seventeen, maar zal in de 2019 gefixed worden, zal ik voor zorgen zodra de homepage verder gevuld wordt. Onze sponsoren zal H1 zijn en de sponsor titels / types zullen H2 zijn, omdat die HTML uit een shortcode komt. 

_* Gedragscode: H2 is missing_
Done: Changed all headers to H2, so no need to introduce a h2 just to have a h2.   Also done for the English version of this page.  
_* Sponsors  H2 is missing.  Done, changed the previous H3 to a h2.  Also: Add a heading H2 sponsors and give gold, silver, bronze a heading H3._
Can't change the heading of the sponsor levels. This is echoed by a shortcode. 
_* Locatie Use H3 sub headings for Openbaar vervoer / Watertaxi etc. Use real list items (it's now a bold dot)_
Done, fixed for NL and EN version. 

# Link text 
Make link text meaningful, standout on their own and never empty. If you use an image as link, use the alt text as link text. For an explanation see: Good link texts

# Link text issues
_*Home: The link texts "here" or "hier" are meaningless. Suggestion: make the whole text, like "Haal hier je ticket" the link text._
Ok, will do.
_* Home: The image with the tickets has no alt text, so the link is empty. Give it, for example the alt text "Haal hier je ticket"._
 Ok, is aangepast.

_* Nieuws. The thumbnail has no alt text but is a link. This results in an empty link text. Give the image the title of the post it's linking to as link text. Note: this maybe a theme issue_
 Yupp, is indeed theme issue. 

_* Never use a full url as link text (screen reader users will thank you) So instead of: https://www.rotterdam.nl/wonen-leven/overzicht-p-r-terreinen/ use overzicht P+R terreinen as link text._
 Ok, changed, thank you.

# Colour contrast 
_The colour contrast should be good, so more people can read the text. See Use of colour. There are many colour contast checkers. An easy one is Colour Contrast Check by Snook._

Done: Made all color ratio 4.5.1 so they are AA.

# Other issues 
An Iframe must have a title attribute, like on the contact page  `<iframe title="Blue City on Google Maps" width="100%" height="450" frameborder="0" scrolling="no" marginheight="0" marginwidth="0" src="[...]"></iframe>`

Sorry, Is generated by shortcode.

# Some advice (not requirements, but good practice)

_* Avoid the use of capitalized sentences or short words in buttons. They are harder to read._
 Done: Changed

_* Avoid the use of centered text, also harder to read. _

On mobile changed it, but not on tablet and higher because of the visual implications.
Can't find a nice alternative.

-------

Again, thank your for the time to do the check!

Greetings from Austria! 🏂
